tracing-opentelemetry-kafka/src/main/java/io/micronaut/tracing/opentelemetry/instrument/kafka/TracingConsumerInterceptor.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.tracing.opentelemetry.instrument.kafka;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerInterceptor;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+
+/**
+ * Default tracing consumer kafka listener.
+ *
+ * @param <K> key class
+ * @param <V> value class
+ *
+ * @since 4.6.0
+ */
+public class TracingConsumerInterceptor<K, V> implements ConsumerInterceptor<K, V> {
```

Comment:
annotate with `@Internal`

Reply:
This is where I'm not sure if you're right: my idea is that we have default implementations of interceptors for kafka with message filtering enabled for tracing. If the user needs to customize the message filtering logic, he will inherit from this filter, override the filterRecord method and configure the application for his interceptors.